environment.yml

```diff
@@ -122,39 +144,48 @@ dependencies:
     - jupyter-core==4.6.3
     - keras-preprocessing==1.1.2
     - kiwisolver==1.2.0
+    - langid==1.1.6
     - lazy-object-proxy==1.4.3
     - markdown==3.2.2
     - markupsafe==1.1.1
     - matplotlib==3.2.2
     - mccabe==0.6.1
     - mistune==0.8.4
+    - mock==2.0.0
     - model-pruning-google-research==0.0.3
+    - monotonic==1.5
     - mpmath==1.1.0
     - nbconvert==5.6.1
     - nbformat==5.0.7
     - notebook==6.1.3
     - numpy==1.19.0
-    - oauth2client==4.1.3
+    - oauth2client==3.0.0
     - oauthlib==3.1.0
     - opt-einsum==3.2.1
     - packaging==20.4
     - pandas==1.0.4
     - pandocfilters==1.4.2
     - parso==0.7.0
+    - pbr==5.5.0
     - pexpect==4.8.0
     - pickleshare==0.7.5
+    - pillow==7.2.0
     - prometheus-client==0.8.0
     - promise==2.3
     - prompt-toolkit==3.0.5
     - protobuf==3.12.2
     - ptyprocess==0.6.0
+    - pyarrow==0.17.1
     - pyasn1==0.4.8
     - pyasn1-modules==0.2.8
     - pycparser==2.20
+    - pydot==1.4.1
     - pygments==2.6.1
     - pylint==2.5.3
+    - pymongo==3.11.0
     - pyparsing==2.4.7
     - pyrsistent==0.16.0
+    - pyspark==3.1.1
```

Comment:
I may remove this later. I'm not totally sure about it. I believe we need pyspark[sql] anyway for pandas UDF functionality.